src/highdicom/sr/templates.py

```diff
@@ -1227,7 +1228,9 @@ def __init__(
         finding_sites: Optional[Sequence[FindingSite]] = None,
         session: Optional[str] = None,
         measurements: Sequence[Measurement] = None,
-        qualitative_evaluations: Optional[Sequence[CodeContentItem]] = None
+        qualitative_evaluations: Optional[
+            Sequence[Union[CodeContentItem, QualitativeEvaluation]]
```

Comment:
A `QualitativeEvaluation` is still a `CodeContentItem`. Therefore, we technically wouldn't need the `Union`.

Reply:
I agree. That's a good idea. It may be slightly confusing if the type annotations differ between the function signature and the docstring, but we can deprecate support of `CodeContentItem` in a future version and only allow `QualitativeEvaluation`. What do you think about adding a deprecation warning?

Reply:
Ok implemented as suggested in 217b4e0 :
- Type annotations just say `CodeContentItem`
- Docstrings just say `QualitativeEvaluation`
- Deprecation warning is raised if you pass a `CodeContentItem`